### PR TITLE
added list_namespaces.  Fixes #165

### LINKED ISSUE #165: pretty-print available schemas
I often find myself looking for a human-readable list of available schemas in jams.  Something that would do the following:

```python
>>> jams.list_schemas()
NAME                          	DESCRIPTION                             
------------------------------------------------------------------------------
beat                          	Beat event markers with optional metrical po..
beat_position                 	Tuples of (position, measure, beats per meas..
blob                          	Arbitrary blob annotations              
chord                         	Expanded chord set from chord_harte.  Includ..
chord_harte                   	Chord annotations following Harte et al., 20..
chord_roman                   	Chord annotations in roman numeral format: t..
key_mode                      	Key and optional mode (major/minor or Greek ..
lyrics                        	Open strings for lyrics annotations     
lyrics_bow                    	Bag of words or n-grams                 
mood_thayer                   	Time-varying emotional measurements as order..
multi_segment                 	Multi-level segmentation. Each interval has ..
note_hz                       	Note pitches in Hz                      
note_midi                     	Note pitches in (fractional) MIDI note numbers
onset                         	Onset event markers.                    
pattern_jku                   	Musical pattern information (MIDI pitch asso..
pitch_class                   	Pitch class in (tonic, pitch class) format
pitch_contour                 	Continuous pitch contours               
pitch_hz                      	[DEPRECATED] Pitch in Hz                
pitch_midi                    	[DEPRECATED] Pitch in (fractional) MIDI note..
segment_open                  	Open label vocabularies allow all strings.
segment_salami_function       	Function label to identify the type of a seg..
segment_salami_lower          	Lowercase label to identify the type of a se..
segment_salami_upper          	Uppercase label to identify the type of a se..
segment_tut                   	Label to identify the type of a segment base..
tag_cal10k                    	CAL10K tags                             
tag_cal500                    	CAL500 tags                             
tag_gtzan                     	GTZAN 10-class genre annotation         
tag_medleydb_instruments      	MedleyDB instrument source annotations  
tag_msd_tagtraum_cd1          	msd tagtraum cd1 13-class genre annotation
tag_msd_tagtraum_cd2          	msd tagtraum cd2 15-class genre annotation
tag_open                      	Open tag vocabularies allow all strings 
tempo                         	Tempo measurements, in beats per minute (BPM)
vector                        	Numerical vector annotations            
```

Here's a one-liner:
```python
print('{:30s}\t{:40s}'.format('NAME', 'DESCRIPTION'))
print('-' * 78)
for sch in sorted(jams.schema.__NAMESPACE__):
    desc = jams.schema.__NAMESPACE__[sch]['description']
    desc = (desc[:44] + '..') if len(desc) > 46 else desc
    print('{:30s}\t{:40s}'.format(sch, desc))
```

But we could easily build this in.

### PR DESCRIPTION
**Pull request by bmcfee**
This PR adds `list_namespaces()`, which pretty-prints the set of available namespaces.

### CODE REVIEW

**Comment by bmcfee** [comment]
As part of this, I think it might be useful to shorten the description fields of namespaces wherever possible to keep things readable.  What do yall think?

File: jams/schemata/namespaces/chord/chord.json [hunk 1]
```diff
@@ -5,6 +5,6 @@
             "pattern": "^((N|X)|(([A-G](b*|#*))((:(maj|min|dim|aug|1|5|sus2|sus4|maj6|min6|7|maj7|min7|dim7|hdim7|minmaj7|aug7|9|maj9|min9|11|maj11|min11|13|maj13|min13)(\\((\\*?((b*|#*)([1-9]|1[0-3]?))(,\\*?((b*|#*)([1-9]|1[0-3]?)))*)\\))?)|(:\\((\\*?((b*|#*)([1-9]|1[0-3]?))(,\\*?((b*|#*)([1-9]|1[0-3]?)))*)\\)))?((/((b*|#*)([1-9]|1[0-3]?)))?)?))$"
         },
         "dense": false,
-        "description": "Expanded chord set from chord_harte.  Includes, for example: X, sus2, 1, 5, and 13."
+        "description": "Expanded chord set. Includes: X, sus2, 1, 5, and 13."
```
Comment (rabitt) [inline jams/schemata/namespaces/chord/chord.json]: remove `.` at the end? You're removing them for the other descriptions.

File: jams/schemata/namespaces/pattern/jku.json [hunk 2]
```diff
@@ -18,6 +18,6 @@
             "required": ["midi_pitch", "morph_pitch", "staff", "pattern_id", "occurrence_id"]
         },
         "dense": false,
-        "description": "Musical pattern information (MIDI pitch associated with a time onset and a duration), including staff height, voice number, occurrence identification, and pattern identification."
+        "description": "Pattern (MIDI pitch), including staff height, voice number, occurrence id, and pattern id."
```
Comment (rabitt) [inline jams/schemata/namespaces/pattern/jku.json]: `.` at the end

Comment (rabitt) [reply jams/schemata/namespaces/pattern/jku.json]: What about `MIDI note number` instead of `MIDI pitch`?

Comment (bmcfee) [reply jams/schemata/namespaces/pattern/jku.json]: Trying to keep it short, I think it's clear enough from context.

**Pull request merged**